### PR TITLE
Fix Caddy download/installs

### PR DESCRIPTION
**Pull request by dowlingw**
#### Pull Request (PR) description
Updates to installation methods:
* Support downloading 2.x releases from GitHub by generating correct release tarball name
* Fix 404 error when using the caddy download service

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -20,15 +20,24 @@
 
   assert_private()
 
+  case $version {
+    /^1\./: {
+      $dl_file_name = "caddy_v${version}_linux_${arch}.tar.gz"
+    }
+    default: {
+      $dl_file_name = "caddy_${version}_linux_${arch}.tar.gz"
```
Comment (igalic) [inline manifests/install.pp:28]: it seems to me that we can easily replace this hard coded Linux, with $kernel

**Review by igalic — COMMENTED** [review]
reminder for the future

**Comment by dowlingw** [comment]
Looks like the caddy download endpoint no longer serves tarballs by default:
```
evil@destroid:~$ wget "https://caddyserver.com/api/download?plugins=http.git,http.filter,http.ipfilter&license=personal&telemetry=off" -qO caddy_linux_amd64
evil@destroid:~$ file caddy_linux_amd64 
caddy_linux_amd64: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, stripped
evil@destroid:~$ 
```

**Comment by dowlingw** [comment]
It occurs to me that the two install methods have different support for the class parameters here:
* Caddy API appears to return the current version only, so would ignore the `$version` parameter
* GitHub releases does not allow customisation of included features / modules or telemetry settings

Given the API appears to be undocumented and the user documentation points people towards the GitHub releases page, I'd suggest deprecating that install method entirely - but that seems really quite heavy handed and a major loss of functionality. Putting some fail checks in place would be a good compromise otherwise.

Have reached out to the folks at Arden labs to clarify whether there's any download api documentation available.

**Comment by dowlingw** [comment]
Sorry folks, I ran out of time to look at this - I'm going to close until I can come back to this one.

**Pull request closed without merging**